my.css

```diff
@@ -16,10 +20,23 @@ body {
   outline: 2px solid green;
 }
 
-.paragraph {
-  font-size:  16px;
+// issue1: nesting
+body {
+  .product {
+    .product-imag {
+      .paragraph {
+        font-size:  1rem;
+      }
+    }
+  }
 }
 
-.layout {
+// over specifiy, creates the same issues as nesting actually
+body .product .product-image .layout {
   float: left;
-}
\ No newline at end of file
+}
+
+// Issue: deadly specificity (!important)
+.product-image {
+  outline: 20px solid red !important;
+}
```

Comment:
Easy rule: Never ever use important. There are only very, very few reasons to use such a monster of specificity. One could be to overrule inline styles `1-0-0-0` vs the higher important specificity `1-0-0-0-0`. But if you can control the content of the inline style: just fix that, don't make it worse.